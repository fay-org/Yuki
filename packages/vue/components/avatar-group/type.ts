import type { NormalSizes } from '../utils'

export const READONLY_AVATAR_KEY = Symbol('avatarKey')

export interface AvatarGroupProvide {
  props: {
    stacked: boolean
    isSquare: boolean
    size: NormalSizes
  }
}
