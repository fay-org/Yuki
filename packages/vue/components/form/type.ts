import { ComputedRef } from 'vue'
import { tuple, NormalSizes } from '../utils'
import type { FormProps } from './props'

export const labelPosition = tuple('left', 'right', 'top')

export type LabelPosition = typeof labelPosition[number]

export const READONLY_FORM_KEY = Symbol('formKey')

export type FormProvide = {
  formProps: FormProps
  getLabelPostion: (postion: LabelPosition | '') => Exclude<LabelPosition, 'top'> | null
}
