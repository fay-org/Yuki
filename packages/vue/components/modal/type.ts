import { Ref, ExtractPropTypes } from 'vue'
import { props, staticModalProps } from './props'

export type ModalProvide = {
  props: ExtractPropTypes<typeof props>
  selfVisible: Ref<boolean>
  setSelfVisible: (val: boolean) => void
}

export const READONLY_MODAL_KEY = 'modalKey'

export type StaticModalOptions = ExtractPropTypes<typeof staticModalProps> & {
  confirm: () => void
  close: () => void
}
