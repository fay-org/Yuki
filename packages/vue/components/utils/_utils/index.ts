export * from './use-expose'
export * from './use-portal'
export * from './use-real-shape'
export * from './use-rect'
export * from './use-resize'
export * from './use-route'
export * from './use-state'
export * from './use-draggable'
export * from './interface'
