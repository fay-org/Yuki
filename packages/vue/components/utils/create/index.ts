export * from './create-name'
export * from './with-install'
export * from './create-bem'