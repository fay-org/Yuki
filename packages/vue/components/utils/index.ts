export * from './create'
export * from './format'
export * from './_utils'
export * from './theme'
export * from './getposition'
export * from './interface'
