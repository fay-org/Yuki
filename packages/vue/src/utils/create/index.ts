export * from './create-name'
export * from './with-install'
export * from './create-bem'
export * from './pick-context'
export * from './create-portal'
export * from './with-modifiers'
