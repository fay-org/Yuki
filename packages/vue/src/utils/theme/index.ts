export * from './color'
export * from './interface'
