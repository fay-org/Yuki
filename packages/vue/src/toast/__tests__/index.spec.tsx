import Toast from '../toast-item'
import { mount } from '@vue/test-utils'
import type { ToastInternalOptions } from '../interface'
import { noop, NormalTypes } from '../../utils'

describe('Toast', () => {
  it('render normal', () => {
    const wrapper = mount(Toast, {
      props: {
        toast: {
          text: 'Test Message',
          visible: true
        } as ToastInternalOptions
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-toast__message').text()).toBe('Test Message')
  })

  it('actions', async () => {
    const wrapper = mount(Toast, {
      props: {
        toast: {
          text: 'Test Message',
          visible: true,
          actions: 'cancel',
          cancel: noop
        } as ToastInternalOptions
      }
    })

    expect(wrapper.find('.fect-toast__action').exists()).toBeTruthy()
    // const actionEl = wrapper.find('.fect-toast__action')
  })

  it('actions custom', async () => {
    const action = () => <div class="action__self">Action</div>

    const wrapper = mount(Toast, {
      props: {
        toast: {
          text: 'Test Message',
          visible: true,
          actions: [action]
        } as ToastInternalOptions
      }
    })
    expect(wrapper.find('.action__self').exists()).toBeTruthy()
    await wrapper.setProps({
      toast: {
        visible: true,
        actions: [action, 'test']
      }
    })
    const actionContainer = wrapper.find('.fect-toast__actions')
    expect(actionContainer.element.childElementCount).toBe(1)
    await wrapper.setProps({
      toast: {
        visible: true,
        actions: []
      }
    })
    expect(wrapper.find('.fect-toast__actions').element.childElementCount).toBe(0)
  })
})
