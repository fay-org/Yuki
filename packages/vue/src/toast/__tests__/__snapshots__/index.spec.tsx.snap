// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast render normal 1`] = `
<transition-stub>
  <div class="fect-toast">
    <div class="fect-toast__message">Test Message</div>
    <div class="fect-toast__actions">
      <!---->
    </div>
  </div>
</transition-stub>
`;
